client/src/app/pages/applications/components/application-detail-drawer/components/download-button.tsx

```diff
@@ -1,56 +1,55 @@
-import React, { useState } from "react";
-import { Alert, Button } from "@patternfly/react-core";
+import React from "react";
+import { Button } from "@patternfly/react-core";
 import spacing from "@patternfly/react-styles/css/utilities/Spacing/spacing";
-import { Application } from "@app/api/models";
-import { Spinner } from "@patternfly/react-core";
-import { useDownloadStaticReport } from "@app/queries/download";
+import { saveAs } from "file-saver";
 
-export enum MimeType {
-  TAR = "tar",
-  YAML = "yaml",
+import { MimeType } from "@app/api/models";
+import { useFetchStaticReport } from "@app/queries/applications";
+import { NotificationsContext } from "@app/components/NotificationsContext";
+import { AxiosError } from "axios";
+import { getAxiosErrorMessage } from "@app/utils/utils";
+
+interface IDownloadButtonProps {
+  id: number;
+  mimeType: MimeType;
 }
-function DownloadButton({
-  application,
+
+export const DownloadButton: React.FC<IDownloadButtonProps> = ({
+  id,
   mimeType,
-}: {
-  application: Application;
-  mimeType: MimeType;
-}) {
-  const {
-    mutate: downloadFile,
-    isLoading,
-    isError,
-  } = useDownloadStaticReport();
+}) => {
```

Comment:
You'll need to pull in the `children` as a prop to have the text (or react node) to render...

```jsx

export const DownloadButton: React.FC<IDownloadButtonProps> = ({
  id,
  mimeType,
  children
}) => {
```

and throw `children: React.ReactNode` in the props interface

Reply:
@sjd78, Yeah I realized that while testing latest, thanks.